airflow/jobs/scheduler_job.py

```diff
@@ -498,19 +499,26 @@ def update_import_errors(session, dagbag):
         session.commit()
 
     @provide_session
-    def create_dag_run(self, dag, session=None):
+    def create_dag_run(self, dag, dag_runs=None, session=None):
         """
         This method checks whether a new DagRun needs to be created
         for a DAG based on scheduling interval.
         Returns DagRun if one is scheduled. Otherwise returns None.
         """
         if dag.schedule_interval and conf.getboolean('scheduler', 'USE_JOB_SCHEDULE'):
-            active_runs = DagRun.find(
-                dag_id=dag.dag_id,
-                state=State.RUNNING,
-                external_trigger=False,
-                session=session
-            )
+            if dag_runs is None:
```

Comment:
Actually why do we need this branch? This is a method on DagFileProcessor, so not really user facing.

Reply:
I did this to facilitate testing of this method both in unit tests and in performance tests.